README.md

```diff
@@ -7,3 +7,291 @@ A highly robust Discord management bot with a featureset designed around content
+```
+
+## VSCode
+You can use any IDE, my choice is [VSCode](https://code.visualstudio.com/)
+
+In VSCode when opening the folder where you've saved the bot code it should recognise that you have a .venv directory and ask if you would like to use this as the interpreter. Choose Yes.
+
+If you don't get this prompt you can [manually select the interpreter](https://code.visualstudio.com/docs/python/environments)
+
+When running the bot you can either manually run it from the termainal using `python ./bot.py` if you add a launch config in vscode we can start a debug session and take advantage of breakpoints.
+
+An example launch.json can be found in the examples folder. This needs to be copied to the root path of the bot code in a directory called '.vscode'. `ROBOBANANA/.vscode/launch.json`
+
+## Required Code changes
+The first thing that we need to do is set up the `config.ini` file.
+Copy `config.ini.example ` to the same directory removing the '.example' part.
+
+### Config.ini required values
+`Token`: This is the token you saved when creating an application.
+
+Anything ending in Channel, or ChannelID will need an ID/Snowflake of a channel. For my personal development I have set this to the value of the default 'General' channel.
+
+With Developer Mode on simply right click the channel name and select **Copy Channel ID**
+
+`BotRoleID`: Go to Roles, you should see a role there with your bots name. right click the role name and copy ID.
+
+`GiftedTier#RoleID and Tier#RoleID`: As above but now with the relevant role id. I used the same ID for the gifted and standard Tier, but if you like you can create more roles for each value.
+
+`ModRoleID`: ID for the mod role.
+
+`GuildID`: Right click your server/server name **Copy Server ID**
+
+Use this same ID for both **GuildID** entries
+
+`MySQL`: Replace the values with what you set when running your mysql server
+```
+[MySQL]
+Username = dbuser1
+Password = dbuser1password
+Host = 0.0.0.0:3306
+Name = testdb
+```
+
+#### Required Source Code Changes
+At time of writing there are a few hard coded values in the file **controllers/sub_controller.py** that needs to be updated to enable the bot to run.
+
+**Line 42**: Update this to be a valid role on your server.
+
+**Lines 123/124/125**: As above update these to be valid role IDs
+
+**Lines 128/129**: Update these IDs to be valid channel IDs from your server.
+
+## Server slash (/) commands.
+[Slash Command Documentation](https://discord.com/developers/docs/interactions/application-commands)
+
+To register custom slash commands with the server you will need to interact with the discord API. In the examples folder there is an example script which will create a / command for creating a prediction.
+
+### Update Script
+- `Application ID`: You can get this ID from the General Information page when creating the application
+- `GUILD ID`: This is the same ID from the config.ini
+- `TOKEN`: The same token used in the config.ini
```

Comment:
This is not strictly true. There's an easier way that syncs all the bot commands. We really need to make this happen automatically on first setup somehow but in the `bot.py#on_ready` method there are four commented out lines. As long as `[Dicord][GuildID]` is set correctly in the config, uncommenting these lines for the first time you run the bot will sync all commands at once. Include a comment recommending to comment these back out after the first run, otherwise devs will get rate limited by Discord when rebooting their bot too many times, which is p annoying.

https://github.com/crscillitoe/RoboBanana/blob/main/bot.py#L65-L68

Reply:
Updated suggesting to use this method. Will be a better way to do it as this way all commands are imported and no guessing at what needs to be configured in that json.